### ISSUE TITLE
Attributes on chained actions not always exposed

### ISSUE THREAD

**Issue by alycejenni** [ckan/ckan#4597]
### CKAN Version if known (or site URL)
Current master branch (latest commit 8f0cb74)

### Issue
**EDIT:** since submitting the issue some things have _stopped_ working so I'm having to change it.

Decorators for chained actions have to be defined in a specific order to ensure that attributes can be used later, e.g.:

This results in the `side_effect_free` attribute not being exposed:
```py
@toolkit.chained_action
@logic.side_effect_free
def datastore_search(prev_func, context, data_dict):
    ...
```

Example of where this needs to be handled: `ckan/default/src/ckan/ckan/views/api.py:263`

```py
side_effect_free = getattr(function, u'side_effect_free', False)  # this returns False for chained actions
```

**Comment by wardi** [comment]
let's come up with a list of action + auth attributes that we'd like to maintain across action function decorators and then copy them all when we wrap an action function. Am I missing any:
```python
ACTION_FUNCTION_ATTRIBUTES = [
    'side_effect_free',
    'auth_audit_exempt',
]
AUTH_FUNCTION_ATTRIBUTES = [
    'auth_sysadmins_check',
    'auth_allow_anonymous_access',
]
```

Also I'm not sure why these functions are wrapping the action functions they decorate instead of adding their attributes directly to the function.

**Comment by iamarnavgarg** [comment]
Hi @wardi I would like to work on this issue?Can you please guide me further in it?

**Comment by bzar** [comment]
Hey, we ran into this when implementing a chained action on `organization_list`. I cobbled together a quick fix for our issue in bzar/ckan@8cc74a. Is there any progress on the more general solution described here?

**Comment by amercader** [comment]
Fixed in #5873
